heufybot/modules/commands/pronouns.py

```diff
@@ -0,0 +1,82 @@
+from twisted.plugin import IPlugin
+from heufybot.moduleinterface import IBotModule
+from heufybot.modules.commandinterface import BotCommand
+from heufybot.utils import networkName
+from zope.interface import implements
+import json, os.path
+
+
+class Pronouns(BotCommand):
+    implements(IPlugin, IBotModule)
+
+    name = "Pronouns"
+
+    def triggers(self):
+        return ["pronouns", "setpron", "rmpron"]
+
+    def actions(self):
+        return super(Pronouns, self).actions() + [ ("pronouns", 1, self.lookUpPronouns) ]
+
+    def lookUpPronouns(self, server, source, user, displayErrors):
+        if not self.bot.moduleHandler.useModuleOnServer(self.name, server):
+            return
+
+        if networkName(self.bot, server) in self.pronstore and user.lower() in self.pronstore[networkName(self.bot,
+                                                                                                          server)]:
+            return {
+                "success": True,
+                "pronouns": self.pronouns[networkName(self.bot, server)][user.lower()]
+            }
+        if displayErrors:
+            error = "User's pronouns have not been specified. Would you like to add your own, {}?".format(user)
+            self.replyPRIVMSG(server, source, error)
+            return {
+                "success": False
+            }
+
+    def load(self):
+        self.help = "Commands: pronouns <user>, setpron <pronouns>, rmpron | "\
+                    "Query the user's pronouns or specify your own."
+        self.commandHelp = {
+            "pronouns": "pronouns <user> | Query user's pronouns",
+            "setpron": "setpron <pronouns> | Set your own pronouns.",
+            "rmpron": "rmpron | Remove your pronouns for a mysterious unspeakable reason"
+        }
+        if "pronouns" not in self.bot.storage:
+            self.bot.storage["pronouns"] = {}
+        self.pronstore = self.bot.storage["pronouns"]
+
+    def checkPermissions(self, server, source, user, command):
+        return True
+
+    def execute(self, server, source, command, params, data):
+        if command == "setpron":
+            if len(params) < 1:
+                self.replyPRIVMSG(server, source, "Your pronouns are ...blank?")
+                return
+            if networkName(self.bot, server) not in self.pronstore:
+                self.pronstore[networkName(self.bot, server)] = {}
+            self.pronstore[networkName(self.bot, server)][data["user"].nick.lower()] = " ".join(params)
+            self.bot.storage["pronouns"] = self.pronstore  # actually unsure why this is needed
+            self.bot.moduleHandler.runGenericAction("pronouns-updated", data["user"].nick, " ".join(params))
```

Comment:
I don't think this action is necessary. The reason location has one is so that chatmap can update accordingly. Right now this creates a method call in the module handler that ultimately does nothing.